napari/layers/utils/spatial_information.py

```diff
@@ -0,0 +1,331 @@
+"""
+This file contains implementation of spatial information for layers.
+
+This information is:
+
+* Scale
+* Translates
+* Rotate
+* Shear
+* Unit
+"""
+
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    overload,
+)
+
+import numpy as np
+import pint
+from psygnal import Signal, SignalGroup
+
+from napari.layers.utils.layer_utils import coerce_affine
+from napari.utils.transforms import Affine, CompositeAffine, TransformChain
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+    UnitsLike = Union[None, str, pint.Unit, dict[str, Union[str, pint.Unit]]]
+    UnitsInfo = Union[None, pint.Unit, dict[str, pint.Unit]]
+
+
+__all__ = ('SpatialInformation',)
+_OPTIONAL_PARAMETERS = {
+    'affine',
+    'axes_labels',
+    'rotate',
+    'scale',
+    'shear',
+    'translate',
+    'units',
+}
+
+
+class SpatialInformationEvents(SignalGroup):
+    affine = Signal(Affine)
+    axes_labels = Signal(tuple)
+    rotate = Signal(np.ndarray)
+    scale = Signal(tuple)
+    shear = Signal(np.ndarray)
+    translate = Signal(tuple)
+    units = Signal(dict)
+
+
+class SpatialInformation:
```

Comment:
I don't like to be the name critic, but I don't love this one.

Mostly because `Spatial` makes me think of physical space (like millimeters) rather general dimensions that could also be temporal or something else. I guess the intent is that space here is supposed to be more general (i.e. with this information you can bring the layer data into some physical vector space with names and units), but that may not be immediately obvious.

Also I don't generally like `Information` appended to class names given that I'd expect all classes to contain some kind of information, so the only effect is to make the name longer.

If we didn't already have `Dims` to track dimensions across layers as well current viewer positions in those dimensions, I'd suggest `Dims` or `Dimensions`. We could consider the longer `LayerDimensions`, but that's quite verbose.

This is optional to address here assuming none of these names are public (see other comment).

Reply:
@andy-sweet @jni @brisvag 
It will be nice to decide about Class name as it will land in multiple docstring/warning etc. 

This class will contain things like:

* scale
* translate
* units
* axes names

For me name `Dims` is wrong, because of the collision of the name. 

Maye name should be something like `SpatailMetadata`, `DimsMetadata`?